src/mono/wasm/runtime/startup.ts

```diff
@@ -928,11 +924,18 @@ export function mono_wasm_set_main_args(name: string, allRuntimeArguments: strin
 /// 1. Emscripten skips a lot of initialization on the pthread workers, Module may not have everything you expect.
 /// 2. Emscripten does not run the preInit or preRun functions in the workers.
 /// 3. At the point when this executes there is no pthread assigned to the worker yet.
-async function mono_wasm_pthread_worker_init(): Promise<void> {
+export async function mono_wasm_pthread_worker_init(): Promise<void> {
+    console.debug("MONO_WASM: worker initializing essential C exports and APIs");
+    // FIXME: copy/pasted from mono_wasm_pre_init_essential - can we share this code? Any other global state that needs initialization?
+    init_c_exports();
+    cwraps_internal(INTERNAL);
```

Comment:
```cwraps_internal(INTERNAL);
cwraps_mono_api(MONO);
cwraps_binding_api(BINDING);
```
should be omitted I think. We are not having public API on worker any time soon.
And when we do, it's not these legacy methods.